File: gitdot-api/src/endpoint/repository/unstar_repository.rs
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;

pub struct UnstarRepository;

impl Endpoint for UnstarRepository {
    const PATH: &'static str = "/repository/{owner}/{repo}/unstar";
    const METHOD: http::Method = http::Method::POST;

    type Request = UnstarRepositoryRequest;
    type Response = UnstarRepositoryResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UnstarRepositoryRequest {}

pub type UnstarRepositoryResponse = ();

File: gitdot-web/packages/gitdot-api/src/endpoint/repository/unstar-repository.ts
import { z } from "zod";
import type { Endpoint } from "../endpoint";

export const UnstarRepositoryRequest = z.object({});
export type UnstarRepositoryRequest = z.infer<typeof UnstarRepositoryRequest>;

export const UnstarRepositoryResponse = z.void();
export type UnstarRepositoryResponse = z.infer<typeof UnstarRepositoryResponse>;

export const UnstarRepository = {
  path: "/repository/{owner}/{repo}/unstar",
  method: "POST",
  request: UnstarRepositoryRequest,
  response: UnstarRepositoryResponse,
} as const satisfies Endpoint;
export type UnstarRepository = typeof UnstarRepository;
